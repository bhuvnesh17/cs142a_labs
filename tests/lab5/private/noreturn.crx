// test functions with value must have return

func validReturn1(a:int) : int {
  return a;
}

func validReturn2(a:int) : int {
  if (true) {
    let a = 3;
  } else {
    return a;
  }
  return 4;
}

func missingReturn1(a:int) : int {
  if (true) {
    return a;
  }
}

func missingReturn2(a:int) : int {
  if (true) {
    return a;
  } else {
    let a = 3;
  }
}

func missingReturn3(a:int) : int {
  if (false) {
    let a = 3;
  } else {
    return a;
  }
}

func validReturn3(a:int) : int {
  while (false) {
    return a;
  }
  return a;
}

func missingReturn4(a:int) : int {
  while(true) {
    return a;
  }
}

func main() : void {
  ::printInt(8);
}

func validReturn4(a:int) : int {
  if (true) {
    return a;
  } else {
    let a = 3;
    return a;
    var b : int;
    let b = 3;
  }
}

func missingReturn5(a:int) : int {
  if (false) {
    let a = 3;
    while(false) {
      return 3;
    }
  } else {
    return a;
  }
}

func missingReturn6(a:int) : int {
  while(true) {
    if (false) {
      let a = 3;
    } else {
      return a;
    }
  }
}

func missingReturn7(a:int) : int {
      var b : int;
      let b = 3;
}


func validReturn5(a:int) : int {
  if (true) {
    return a;
  } else {
  }
  return a;
}

func missingReturn8(a:int) : int {
  if (true) {
    return a;
  } else {
  }
}

func missingReturn9(a:int) : int {
  while(true) {
    if (true) {
      return a;
    } else {
    }
  }
}

func validReturn6(a:int) : int {
  if (true) {
    return a;
  } else {
  }
  return a;
  while(true) {
    return a;
  }
}

func missingReturn10(a:int) : int {
  while(true) {
    if (true) {
    } else {
      return a;
    }
  }
}

func validReturn7(a:int) : int {
  if (true) {
    return a;
    return a;
    return a;
    return a;
  } else {
    while (true) {
      return a;
      var b : int;
      let b = 3;
      return b;
      return b;
      return b;
      return b;
    }
      return a;
      return a;
      return a;
      return a;
      return a;
  }
  return a;
  return a;
  return a;
  return a;
}

func validReturn8(a:int) : int {
  if (true) {
    let a = 3;
  } else {
  }

  if (false) {
    return a;
  } else {
    return a;
  }
}

func missingReturn11(a:int) : int {
  if (true) {
    return a;
  } else {
  }

  if (false) {
  } else {
    return a;
  }
}

func missingReturn12(a:int) : int {
  if (true) {
    let a = 8;
  } else {
  }

  if (false) {
  } else {
    return a;
  }
}
