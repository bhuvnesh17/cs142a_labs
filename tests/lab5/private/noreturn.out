Error type-checking file.
TypeError(16,1)[Not all paths in function missingReturn1 have a return.]
TypeError(22,1)[Not all paths in function missingReturn2 have a return.]
TypeError(30,1)[Not all paths in function missingReturn3 have a return.]
TypeError(45,1)[Not all paths in function missingReturn4 have a return.]
TypeError(66,1)[Not all paths in function missingReturn5 have a return.]
TypeError(77,1)[Not all paths in function missingReturn6 have a return.]
TypeError(87,1)[Not all paths in function missingReturn7 have a return.]
TypeError(101,1)[Not all paths in function missingReturn8 have a return.]
TypeError(108,1)[Not all paths in function missingReturn9 have a return.]
TypeError(128,1)[Not all paths in function missingReturn10 have a return.]
TypeError(178,1)[Not all paths in function missingReturn11 have a return.]
TypeError(190,1)[Not all paths in function missingReturn12 have a return.]

